PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTrackMatcher.cxx

```diff
@@ -0,0 +1,252 @@
+// AliEmcalCorrectionCellTrackMatcher
+//
+
+#include "AliEmcalCorrectionCellTrackMatcher.h"
+
+#include <iostream>
+
+#include "AliClusterContainer.h"
+#include "AliParticleContainer.h"
+#include "AliEMCALRecoUtils.h"
+#include "AliESDCaloCluster.h"
+#include "AliAODCaloCluster.h"
+#include "AliVParticle.h"
+#include "AliEmcalParticle.h"
+#include "AliEMCALGeometry.h"
+#include "AliMCEvent.h"
+
+
+
+/// \cond CLASSIMP
+ClassImp(AliEmcalCorrectionCellTrackMatcher);
+/// \endcond
+
+// Actually registers the class with the base class
+RegisterCorrectionComponent<AliEmcalCorrectionCellTrackMatcher> AliEmcalCorrectionCellTrackMatcher::reg("AliEmcalCorrectionCellTrackMatcher");
+
+/**
+ * Default constructor
+ */
+AliEmcalCorrectionCellTrackMatcher::AliEmcalCorrectionCellTrackMatcher() :
+  AliEmcalCorrectionComponent("AliEmcalCorrectionCellTrackMatcher"),
+  fMipE(290.),
+  fEmcalTracks(0),
+  fCellTrackMatchdEtadPhi(0),
+  fCellNTrackMatch(0),
+  fCellTrackMatchEbefore(0),
+  fCellTrackMatchEafter(0)
+{
+}
+
+/**
+ * Destructor
+ */
+AliEmcalCorrectionCellTrackMatcher::~AliEmcalCorrectionCellTrackMatcher()
+{
+}
+
+/**
+ * Initialize and configure the component.
+ */
+Bool_t AliEmcalCorrectionCellTrackMatcher::Initialize()
+{
+  // Initialization
+  AliEmcalCorrectionComponent::Initialize();
+
+  GetProperty("mipE", fMipE);
+
+  return kTRUE;
+}
+
+/**
+ * Create run-independent objects for output. Called before running over events.
+ */
+void AliEmcalCorrectionCellTrackMatcher::UserCreateOutputObjects()
+{
+  AliEmcalCorrectionComponent::UserCreateOutputObjects();
+
+  // Determine all particle container array names for naming the AliEmcalParticle array
+  std::string particleContainerNames = "";
+  bool firstLoop = true;
+  AliParticleContainer * partCont = 0;
+  TIter nextPartCont(&fParticleCollArray);
+  while ((partCont = static_cast<AliParticleContainer*>(nextPartCont()))) {
+    if (firstLoop != true) {
+      particleContainerNames += "_";
+    }
+    else {
+      firstLoop = false;
+    }
+    particleContainerNames += partCont->GetArrayName();
+  }
+
+  fEmcalTracks = new TClonesArray("AliEmcalParticle");
+  fEmcalTracks->SetName(Form("EmcalTracks_%s", particleContainerNames.c_str()));
```

Comment:
In fact, I think all of this stuff can be removed.

Now I think I see what's going on - all of this info is only necessary if you're storing the matching information. But as far as I can tell, you're not, so it's not necessary